apps/coscrad-frontend/src/components/resources/spatial-features/utils/find-first-point.ts

```diff
@@ -0,0 +1,26 @@
+import { Position2D, SpatialFeatureCoordinatesUnion } from '../types';
+
+type GeometricCoordinates = SpatialFeatureCoordinatesUnion;
+
+/**
+ * Returns the first point on a `Point`, `Line`, or `MultiPolygon`.
+ *
+ * Returns `undefined` if an empty array is passed in.
+ **/
+export const findFirstPoint = (coordinates: GeometricCoordinates): Position2D => {
+    if (coordinates.length === 0) return undefined;
+
+    const testElement = coordinates[0];
+
+    // we have a point
+    if (typeof testElement === 'number') return coordinates as Position2D;
+
+    const nestedTestElement = testElement[0];
+
+    if (typeof nestedTestElement === 'number') {
+        // testElement's 0th member is a number so `testElement` is a point
```

Comment:
Not sure I understand this.  After line 16, `testElement` is an array rather than just a number.  So then `nestedTestElement` is the 0th member of that array.  Oh, I see, on 16, `coordinates` is a `Point`, then on 21, `testElement` is a `Point` and `coordinates` is an array of `Points`.  Then after 23, what's left?  Is `nestedTestElement` an array?  It doesn't need a type assertion?  Just trying to reason through this.

Reply:
Oh, is it then a `ring`?  An array of points?  Seems simple but my brain is still exploding...